src/main/java/org/example/hospital/service/MessageService.java

```diff
@@ -0,0 +1,92 @@
+package org.example.hospital.service;
+
+import org.example.hospital.dto.MessageDTO;
+import org.example.hospital.entity.Message;
+import org.example.hospital.repository.MessageRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+public class MessageService {
+    private final MessageRepository messageRepository;
+    private final UserService userService;
+
+    @Autowired
+    public MessageService(MessageRepository messageRepository, UserService userService) {
+        this.messageRepository = messageRepository;
+        this.userService = userService;
+    }
+
+    @Transactional
+    public void addMessage(Long receiverId, MessageDTO messageDTO){
+        Message message = new Message();
+        message.setSender(userService.getUserById(messageDTO.getSenderId()));
```

Comment:
Change this to userService.getUserReferenceById and add a method in UserService to avoid non-nesessary query on user.